SessionD build warnings seen on master

Seeing this on master this morning (jul 27) when running `make build_session_manager`
```
In file included from /home/vagrant/magma/lte/gateway/c/session_manager/SessionState.cpp:29:
/home/vagrant/magma/lte/gateway/c/session_manager/SessionState.h: In constructor ‘magma::SessionState::SessionState(const magma::StoredSessionState&, magma::StaticRuleStore&)’:
/home/vagrant/magma/lte/gateway/c/session_manager/SessionState.h:802:12: warning: ‘magma::SessionState::shard_id_’ will be initialized after [-Wreorder]
  802 |   uint16_t shard_id_;
      |            ^~~~~~~~~
/home/vagrant/magma/lte/gateway/c/session_manager/SessionState.h:740:12: warning:   ‘uint32_t magma::SessionState::request_number_’ [-Wreorder]
  740 |   uint32_t request_number_;
      |            ^~~~~~~~~~~~~~~
/home/vagrant/magma/lte/gateway/c/session_manager/SessionState.cpp:134:1: warning:   when initialized here [-Wreorder]
  134 | SessionState::SessionState(
      | ^~~~~~~~~~~~
[59/59] Linking CXX executable sessiond
```

--- a/lte/gateway/c/session_manager/SessionState.cpp
+++ b/lte/gateway/c/session_manager/SessionState.cpp
@@ -135,7 +135,6 @@ SessionState::SessionState(
     const StoredSessionState& marshaled, StaticRuleStore& rule_store)
     : imsi_(marshaled.imsi),
       session_id_(marshaled.session_id),
-      shard_id_(marshaled.shard_id),
       request_number_(marshaled.request_number),
       curr_state_(marshaled.fsm_state),
       config_(marshaled.config),
@@ -153,7 +152,8 @@ SessionState::SessionState(
       pending_event_triggers_(marshaled.pending_event_triggers),
       revalidation_time_(marshaled.revalidation_time),
       credit_map_(4, &ccHash, &ccEqual),
-      bearer_id_by_policy_(marshaled.bearer_id_by_policy) {
+      bearer_id_by_policy_(marshaled.bearer_id_by_policy),
+      shard_id_(marshaled.shard_id) {
   session_level_key_ = marshaled.session_level_key;
   for (auto it : marshaled.monitor_map) {
     Monitor monitor;

